compiler/resolution/functionResolution.cpp

```diff
@@ -7334,11 +7334,13 @@ static Type* moveDetermineRhsType(CallExpr* call) {
         // `expandExternArrayCalls` can add void assignments when
         // the extern function has an explicit void return type.
         // Let those through by looking for two flags it adds.
-        if (!(rhsFn->hasFlag(FLAG_VOID_NO_RETURN_VALUE) &&
+        if (!rhsFn->hasFlag(FLAG_INIT_COPY_FN) &&
```

Comment:
This function resolves moves of the form `lhsFn = rhsFn`. If the `FLAG_INIT_COPY_FN` flag is set for a `rhsFn` returning `void`, such as when `void` aggregate fields are default-initialized, we ignore the use of the function's return value, and do not flag it as an `illegal use of a function that does not return a value`.